pkg/hub/submarineragent/controller.go

```diff
@@ -785,3 +785,11 @@ func (c *submarinerAgentController) createGNConfigMapIfNecessary(brokerNamespace
 
 	return nil
 }
+
+func isSpokePrepared(cloudName string) bool {
+	if cloudName == "GCP" || cloudName == "OpenStack" || cloudName == "Azure" {
+		return true
+	}
+
+	return false
```

Comment:
I believe only AWS is done on the hub, if so:

```suggestion
	return cloudName != "AWS"
```

Reply:
But since any thing other that does not use submariner cloud prepare (like vmware ) need to take a different path,

Reply:
Not sure I follow.

Reply:
I mean to say there are three cases
if it is AWS it will take this path 
https://github.com/stolostron/submariner-addon/blob/main/pkg/spoke/submarineragent/config_controller.go#L174

if it s GCP, Azure  or Openstack  it takes this path
https://github.com/stolostron/submariner-addon/blob/main/pkg/spoke/submarineragent/config_controller.go#L180

if it is VMware or any other cloud it takes this path
https://github.com/stolostron/submariner-addon/blob/main/pkg/spoke/submarineragent/config_controller.go#L185

is it is not AWS we need to check if it is case 2 or case3

Reply:
ah oh ok got it it was my bad. I was just thinking it is config controller in spoke

Reply:
Done.